registrace_bp.py

```diff
@@ -1,73 +1,82 @@
+    city = StringField(
+        'Město/Obec*:', [validators.Required(message='Vyplň město, kde bydlíš.')])
+    postcode = IntegerField('PSČ*:', [validators.Required(message='Bez PSČ to nepůjde.'),
+                                      validators.Length(min=5, max=5, message='PSČ zadej bez mezer.')])
+    email = StringField('E-mail*:', [validators.Required(message='Bez mailu to nepůjde.', ),
+                                     validators.Email(message='Chybný tvar emailové adresy.'), validators.Length(min=4, max=250)])
+    phone = IntegerField('Telefon*:', [validators.Required(message='Bez telefonního čísla to nepůjde.'), '''validators.Length(
+        min=11, max=11, message='Telefonní číslo zadej bez předvolby a ve tvaru xxx xxx xxx.')'''])
+    password = PasswordField('Heslo*:', [validators.Required(message='Bez hesla to nepůjde.'), validators.EqualTo(
+        'confirm_password', message='Hesla se musejí shodovat.'), validators.Length(min=8, message='Heslo musí být alespoň 6 znaků dlouhé.')])
+    confirm_password = PasswordField(
+        'Potvrzení hesla*:', [validators.Required(message='Zadané heslo je třeba potvrdit.')])
+    gdpr = BooleanField('Souhlasím ze zpracováním osobních údajů.*', [validators.Required(
+        message='Bez udělení Tvého souhlasu Tě nemůžeme zaregistrovat.')])
+
+
 blueprint = Blueprint('registrace_bp', __name__)
+
+
 @blueprint.route('/registrace')
 def show_registrace():
-	zavody = db_funkce.zavody()
-	uzivatel = current_user
-	if uzivatel.is_authenticated:
-		flash ('Už jsi přihlášen.', "danger")
-		return render_template ('zavody.html',  zavody=zavody, id_vybraneho=0)
-	return render_template('registrace.html', values={})
+    uzivatel = current_user
+    form = RegistrationForm()
 
-@blueprint.route('/registrace', methods=['POST'])
+    if uzivatel.is_authenticated:
+        flash('Už jsi přihlášen.', "danger")
+        return redirect(url_for('zavody_bp.show_zavody'))
+    return render_template('registrace.html', values={}, form=form)
+
+
+@blueprint.route('/registrace', methods=['POST', 'GET'])
 def add_new():
-	result = request.form
-	chyba = None
+    result = request.form
+    chyba = None
+    form = RegistrationForm()
+
+    if form.validate_on_submit():
+        uz_registrovany = db_funkce.najdi_uzivatele(result.get("email"))
+        if uz_registrovany:
+            flash('Už jsi u nás byl/a, tak se prosím přihlaš.', "danger")
+            return redirect(url_for('prihlaseni_bp.login'))
 
-	uz_registrovany = db_funkce.najdi_uzivatele(result.get("email"))
-	if uz_registrovany:
-		flash ('Už jsi u nás byl/a, tak se prosím přihlaš.', "danger")
-		return redirect(url_for('prihlaseni_bp.login'))
+        heslo = result.get("password")
+        heslo_potvrzeni = result.get("confirm_password")
+        if not heslo == heslo_potvrzeni:
+            flash('Hesla se neshodují.', "danger")
+            return render_template("registrace.html", values=result, error=chyba, form=form)
```

Comment:
Ty `values` jsou teď už nevyužité, pokud vím, tak bych to odsud klidně vyhodil (stejně jako `error`)